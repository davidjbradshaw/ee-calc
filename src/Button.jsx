import React from 'react'
import PropTypes from 'prop-types'

import { eeBlue } from './consts'

const buttonStyle={
  backgroundColor: 'white',
  color: eeBlue,
  borderRadius: '5px',
  border: 'solid 2px white',
  height: '40px',
  width: '40px',
  margin: '0 0 5px 5px',
  textAlign: 'centre',
  fontSize: '20px'
}

const Button = ({value, onClick}) => (
  <button type="button" value={value} onClick={() => onClick(value)} style={buttonStyle}>
    {value}
  </button>
)

Button.prototypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]).isRequired,
  onClick: PropTypes.func.isRequired
}

export default Button