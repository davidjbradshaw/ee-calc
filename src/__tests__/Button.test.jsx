import React from 'react'
import { identity } from 'ramda'
import { shallow } from 'enzyme'
import Button from '../Button'

it('Renders a button', () => {
  const value = "foo"
  const button = <Button value={value} onClick={identity}/>
  expect(button).toMatchSnapshot()

  const component = shallow(button)
  expect(component.invoke('onClick')()).toBe(value)
  expect(component.prop('value')).toBe(value)
})
