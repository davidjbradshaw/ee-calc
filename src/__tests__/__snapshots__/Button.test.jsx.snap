// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a button 1`] = `
<Button
  onClick={[Function]}
  value="foo"
/>
`;
